pkg/local-disk-manager/controller/localdisk/localdisk_controller.go

```diff
@@ -112,29 +104,154 @@ func (r *ReconcileLocalDisk) Reconcile(_ context.Context, req reconcile.Request)
+	err := r.diskHandler.UpdateStatus()
+	if err != nil {
+		eventReason = v1alpha1.LocalDiskEventReasonReservedFail
+		eventType = v1.EventTypeWarning
+		eventMessage = fmt.Sprintf("Reserve Disk %v failed", disk.GetName())
+	}
+
+	r.Recorder.Eventf(disk, eventType, eventReason, eventMessage)
+	return err
+}
+
+// reservedDisk update disk status to Bound
+func (r *ReconcileLocalDisk) updateDiskStatusBound(disk *v1alpha1.LocalDisk) error {
+	var (
+		eventReason  = v1alpha1.LocalDiskEventReasonBound
+		eventType    = v1.EventTypeNormal
+		eventMessage = fmt.Sprintf("Bound Disk %v succeed", disk.GetName())
+	)
+
+	r.diskHandler.SetupStatus(v1alpha1.LocalDiskBound)
+	err := r.diskHandler.UpdateStatus()
+	if err != nil {
+		eventReason = v1alpha1.LocalDiskEventReasonBoundFail
+		eventType = v1.EventTypeWarning
+		eventMessage = fmt.Sprintf("Bound Disk %v failed", disk.GetName())
 	}
 
-	// NOTE: The control logic of localdisk should only respond to the update events of the disk itself.
-	// For example, if the disk smart check fails, it should update its health status at this time.
-	// As for the upper layer resources that depend on it, such as LDC, what it should do is to monitor
-	// the event changes of LD and adjust the changed contents accordingly.
-	// The connection between them should only be related to the state.
-
-	// Update status
-	if ldHandler.ClaimRef() != nil && ldHandler.UnClaimed() {
-		ldHandler.SetupStatus(v1alpha1.LocalDiskClaimed)
-		if err := ldHandler.UpdateStatus(); err != nil {
-			r.Recorder.Eventf(&ldHandler.Ld, v1.EventTypeWarning, "UpdateStatusFail", "Update status fail, due to error: %v", err)
-			log.WithError(err).Errorf("Update LocalDisk %v status fail", ldHandler.Ld.Name)
-			return reconcile.Result{}, err
+	r.Recorder.Eventf(disk, eventType, eventReason, eventMessage)
+	return err
+}
+
+// processDiskEmpty update disk status to Available
+func (r *ReconcileLocalDisk) processDiskReserved(disk *v1alpha1.LocalDisk) error {
+	logCtx := log.Fields{"name": disk.Name}
+	log.WithFields(logCtx).Info("Start to processing Reserved localdisk")
+
+	var (
+		err error
+	)
+
+	// Update to Bound if detect already be in used
+	if disk.Spec.HasPartition {
+		r.diskHandler.SetupStatus(v1alpha1.LocalDiskBound)
+		if err = r.diskHandler.UpdateStatus(); err == nil {
+			log.WithFields(logCtx).Info("Update a Reserved disk state to Bound")
+			r.Recorder.Eventf(disk, v1.EventTypeNormal, "Update disk %v state from Reserved to Bound, "+
+				"due to detect partition or filesystem on it", disk.Name)
+		}
+	}
+
+	return err
+}
+
+// processDiskEmpty update disk status to Available
+func (r *ReconcileLocalDisk) processDiskBound(disk *v1alpha1.LocalDisk) error {
+	logCtx := log.Fields{"name": disk.Name}
+	log.WithFields(logCtx).Info("Start to processing Bound localdisk")
+
+	var (
+		err error
+	)
+
+	// Check if disk can be released
+	if disk.Spec.ClaimRef != nil && disk.Spec.HasPartition {
```

Comment:
why can the disk be released when it has ClaimRef or HasPartition?

Reply:
Seems mistake, it should use !HasPartition